Dockerfile

```diff
@@ -501,6 +501,96 @@ RUN \
 
 # =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
 
+FROM sdk-libc as sdk-go-fips
+
+ARG ARCH
+ARG TARGET="${ARCH}-bottlerocket-linux-gnu"
+ARG SYSROOT="/${TARGET}/sys-root"
+ARG GOVER="1.21.4"
+ARG AWS_LC_FIPS_VER="2.0.2"
+
+USER root
+RUN DEBIAN_FRONTEND=noninteractive dnf install -y golang cmake xz wget ca-certificates clang python curl git
+
+USER builder
+WORKDIR /home/builder/sdk-go-fips
+COPY ./hashes/go /home/builder/hashes
+RUN \
+  sdk-fetch /home/builder/hashes && \
+  tar --strip-components=1 -xf go${GOVER}.src.tar.gz && \
+      rm go${GOVER}.src.tar.gz
+
+WORKDIR /home/builder/sdk-go-fips
+COPY patches/go-fips/* ./
+RUN git init && git apply --whitespace=nowarn *.patch
+
+# Build aws-lc crypto modules
+USER root
+WORKDIR /boring/boringssl
+COPY ./hashes/aws-lc /home/builder/hashes
+ENV UPSTREAM_SOURCE_FALLBACK=true
+RUN \
+  sdk-fetch /home/builder/hashes && \
+  tar --strip-components=1 -xf AWS-LC-FIPS-${AWS_LC_FIPS_VER}.tar.gz && \
+    rm AWS-LC-FIPS-${AWS_LC_FIPS_VER}.tar.gz
+RUN \
+  install -p -m 0644 -Dt /aws-lc/licenses LICENSE
```

Comment:
The final install step can just pull this from the `/home/builder/...` directory - we don't want to use directories under `/` since they pull toward doing the build as root.